mlflow/langchain/__init__.py

```diff
@@ -151,6 +158,13 @@ def save_model(
 
                      .. Note:: Experimental: This parameter may change or be removed in a future
                                              release without warning.
+    :param loader_fn: A function that takes a string `persist_dir` defined below as the argument
+                      and returns a retriever. This is required for RetrievalQA models.
+    :param persist_dir: The directory where the retriever is stored. The `loader_fn` takes this
```

Comment:
I'm trying to understand how this argument works. Let's say we have two machines, A and B. You train and log a model on A with `persist_dir="/path/to/persist/dir"`, then try to serve it on B. Do we need to populate `/path/to/persist/dir` in B?

Reply:
Yes, you do. One scenario is to set the `persist_dir` to a DBFS FUSE location so that it is available on both A and B.

Reply:
Sorry, I must have missed that we're not downloading `persist_dir` and storing it with the model. Unless we do that, the model won't be deployable to Databricks Model Serving, can't be registered with UC Model Registry, etc.

@liangz1 Can we download `persisit_dir` and store with the model? Can we also add a test case verifying that the model can be loaded even after the original `persist_dir` is deleted?

Reply:
Can a retriever be large?

Reply:
@dbczumar You are not missing it. I did say "The user should persist the DB, log the DB files and the code to load the persisted files back as artifacts in MLflow." Torn between option A (uploading persist_dir) and option B (not uploading persist_dir), I was in a mental whirlwind. In the confusion, I ended up designing it as A but implementing B instead.

It sounds like even though the persist_dir could be large, we still want to upload it as artifacts. Right?

Reply:
Definitely, we want to upload :). The persist dir is going to be waaaay smaller than most LLMs :D

Reply:
Good point!